File: src/main/java/com/thesilentnights/easylogin/service/AccountService.java
package com.thesilentnights.easylogin.service;

import com.thesilentnights.easylogin.pojo.PlayerAccount;
import com.thesilentnights.easylogin.sql.DataSource;

import java.util.Optional;
import java.util.UUID;

public class AccountService {

    private static DataSource dataSource;

    public static void init(DataSource dataSource) {
        AccountService.dataSource = dataSource;
    }

    public static boolean hasAccount(UUID uuid) {
        return dataSource.getAuthByUUID(uuid).isPresent();
    }

    public static boolean hasAccount(String username) {
        return dataSource.getAuthByName(username).isPresent();
    }

    public static Optional<PlayerAccount> getAccount(UUID uuid) {
        return dataSource.getAuthByUUID(uuid);
    }

    public static Optional<PlayerAccount> getAccount(String username) {
        return dataSource.getAuthByName(username);
    }

    public static boolean updatePassword(String value, UUID uuid) {
        return dataSource.updatePassword(value, uuid);
    }

    public static void updateAccount(PlayerAccount account) {
        if (hasAccount(account.getUuid())) {
            dataSource.updateAccount(account);
        } else {
            dataSource.insertAccount(account);
        }
    }
}

File: src/main/kotlin/com/thesilentnights/easylogin/service/AccountService.kt
package com.thesilentnights.easylogin.service

import com.thesilentnights.easylogin.pojo.PlayerAccount
import com.thesilentnights.easylogin.pojo.SqlColumnDefinition
import com.thesilentnights.easylogin.sql.DataSource
import java.util.*


object AccountService {
    private lateinit var dataSource: DataSource
    fun init(dataSource: DataSource) {
        this.dataSource = dataSource
    }

    fun hasAccount(uuid: UUID): Boolean {
        return dataSource.getAuthByUUID(uuid).isPresent
    }

    fun hasAccount(username: String?): Boolean {
        return dataSource.getAuthByName(username).isPresent
    }

    fun getAccount(uuid: UUID): Optional<PlayerAccount> {
        return dataSource.getAuthByUUID(uuid)
    }

    fun getAccount(username: String): Optional<PlayerAccount> {
        return dataSource.getAuthByName(username)
    }

    fun updateSingleColumn(key: SqlColumnDefinition, value: String, uuid: UUID): Boolean {
        return dataSource.updateColumn(key, value, uuid)
    }

    fun updateAccount(account: PlayerAccount) {
        if (hasAccount(account.uuid)) {
            dataSource.updateAccount(account)
        } else {
            dataSource.insertAccount(account)
        }
    }
}
